### PR TITLE
Wam2c

### PR DESCRIPTION
**Pull request by WChen-NOAA**
# PR Checklist

- [ ] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [ ] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [ ] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Provide a detailed description of what this PR does. What bug does it fix, or what feature does it add? Is a change of answers expected from this PR? Are any library updates included in this PR (modulefiles etc.)?

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [ ] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/nems/pull/<pr_number>
- waiting on noaa-emc/fv3atm/pull/<pr_number>

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
Good morning @WChen-NOAA . Does this PR bring the NOAA-SWPC fork of UFS-Weather-Model up to date with the NOAA-EMC develop branch (or close to it)?

**Comment by BrianCurtis-NOAA** [comment]
There are a lot of changes being brought into this PR, but because I suspect it's from bringing the branch up-to-date (or close to) the upstream repo, it's beyond a need to review every change.

Once this merge is made, then as the NOAA-EMC PR's are created It will become easy to see what this work is specifically changing and I will spend a lot of time reviewing those.

I've run the 4 new RT's on Hera with your input directory and they pass successfully. I see no reason at this point (for me) to hold this work up any more. Approved.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by WChen-NOAA** [comment]
Hi Brain,
Thank you very much for helping. 
I think the code is close to the main branch of UFS-WM but all the updates were done before I took my family leave.
And I think it needs to be hold for a while before making the PR to the main branch of UFS-WM since I still have some concerns on the updates of other modules such as MOM.

**Pull request closed without merging**
